pkg/terra/db/db.go

```diff
@@ -60,14 +60,17 @@ type State string
 
 var (
 	// Unstarted means queued but not processed.
-	// Valid next states: Broadcasted, Errored (sim fails)
+	// Valid next states: Broadcasted, Errored (sim fails), Replaced (superseded)
 	Unstarted State = "unstarted"
 	// Broadcasted means included in the mempool of a node.
 	// Valid next states: Confirmed (found onchain), Errored (tx expired waiting for confirmation)
 	Broadcasted State = "broadcasted"
 	// Confirmed means we're able to retrieve the txhash of the tx which broadcasted the msg.
 	// Valid next states: none, terminal state
 	Confirmed State = "confirmed"
+	// Replaced means the caller sent another message which superseded this one before it was broadcast.
+	// Valid next states, none, terminal state. However, a mid-broadcast replacement attempt could be overwritten.
+	Replaced State = "replaced"
```

Comment:
I'm thinking this would be simpler and more flexible going forward if generalized, e.g. `cancelled`, with the particular case of replacement being an implementation detail.
```suggestion
	Cancelled State = "cancelled"
```